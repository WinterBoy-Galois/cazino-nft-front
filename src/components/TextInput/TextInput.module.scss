@import '../../styles/shared';

.inputField {
  text-align: left;
  background-color: $form-background;
  color: $white;
  font-size: 18px;
  line-height: 24px;
  margin-left: 0;
  padding: 0;
  transition: all 0.2s ease-in;
  outline: none;
  border: none;

  &:disabled {
    background-color: $read-only-input-bg;
  }

  &__container {
    width: 100%;
    display: flex;
    flex-direction: column;
    margin-bottom: 1rem;
  }

  &__wrapper {
    flex: 1;
    display: inline-block;
    padding: 5px 10px 5px 10px;
    border: 2px solid $form-background;
    border-radius: 8px;
    background-color: $form-background;
    height: 57px;
    vertical-align: bottom;
    transition: all 0.2s ease-in;

    &--disabled {
      background-color: $read-only-input-bg;
      border-color: $read-only-input-bg;
    }

    &:focus-within {
      border: 2px solid $bitcoinYellow;
    }

    &--error {
      border: 2px solid $red;

      &:focus-within {
        border: 2px solid $red;
      }
    }

    input[type='text'].inputField {
      size: '';
      width: 100%;
    }

    input {
      filter: none;
    }

    input:-webkit-autofill,
    input:-webkit-autofill:hover,
    input:-webkit-autofill:focus {
      border: none;
      -webkit-text-fill-color: $white;
      -webkit-box-shadow: 0 0 0 1000px $form-background inset;
      box-shadow: 0 0 0 1000px $form-background inset;
      transition: background-color 5000s ease-in-out 0s;
    }
  }

  &__label {
    text-transform: uppercase;
    display: block;
    font-size: 14px;
    line-height: 19px;
    color: $live-updates-light-blue;
    user-select: none;
    margin: 0;
  }

  &__error {
    text-transform: uppercase;
    display: block;
    font-size: 14px;
    line-height: 19px;
    color: $red;
    user-select: none;
    margin: 4px 8px 4px 8px;
  }
}
