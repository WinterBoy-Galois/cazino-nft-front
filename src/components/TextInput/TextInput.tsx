import React, { useState, useEffect } from 'react';

import styles from './TextInput.module.scss';

interface IProps {
  name?: string;
  value?: string;
  label?: string;
  validationMessage?: string;
  onChangeValue?: (value: string) => void;
  onBlur?: ({ target }: { target: EventTarget | null }) => void;
}

const TextInput = ({
  name = undefined,
  value: initialValue = undefined,
  label = undefined,
  validationMessage = undefined,
  onChangeValue = undefined,
  onBlur = undefined,
}: IProps) => {
  const [value, setValue] = useState(initialValue);

  useEffect(() => {
    setValue(initialValue);
  }, [initialValue]);

  const isError = () => {
    return !!validationMessage;
  };

  const handleOnChange = (e: { target: { value: string } }) => {
    const newValue = e.target.value;

    setValue(newValue);
    if (onChangeValue) {
      onChangeValue(newValue);
    }
  };

  const keypressHandler = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      // refAmount?.current?.blur();
    }
  };

  const handleBlur = (event?: React.FocusEvent<HTMLInputElement>) => {
    if (onBlur && event) {
      onBlur(event);
    }
  };

  return (
    <div className={styles.inputField__container}>
      <div
        className={
          !isError()
            ? `${styles['inputField__wrapper']}`
            : `${styles['inputField__wrapper']} ${styles['inputField__wrapper--error']}`
        }
      >
        <label className={styles.inputField__label}>{label}</label>
        <input
          {...(name ? { name: name } : {})}
          type="text"
          value={value}
          className={styles.inputField}
          autoComplete="off"
          onChange={handleOnChange}
          onKeyPress={keypressHandler}
          onBlur={handleBlur}
        />
      </div>
      {isError() && <div className={styles.inputField__error}>{validationMessage}</div>}
    </div>
  );
};

export default TextInput;
