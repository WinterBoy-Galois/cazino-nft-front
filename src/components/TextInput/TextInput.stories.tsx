import React from 'react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text } from '@storybook/addon-knobs';

import TextInput from '.';

export default {
  title: 'Components/Forms/TextInput',
  component: TextInput,
  decorators: [
    withKnobs,
    (storyFn: () => React.ReactNode) => <div style={{ padding: '1rem' }}>{storyFn()}</div>,
  ],
  parameters: {
    backgrounds: {
      default: 'Secondary',
    },
  },
};

const data = {
  label: 'Username',
  value: 'sergioalvarez',
  validationMessage: 'Username already exists',
  onChangeValue: action('onChangeValue'),
};

export const Default = () => <TextInput {...data} validationMessage={undefined} />;

export const ValidationError = () => <TextInput {...data} />;

export const Custom = () => (
  <TextInput
    label={text('Label', data.label)}
    value={text('Value', data.value)}
    validationMessage={text('Validation', data.validationMessage)}
  />
);
