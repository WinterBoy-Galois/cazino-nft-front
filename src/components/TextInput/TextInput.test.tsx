import React from 'react';
import { render } from '@testing-library/react';

import TextInput from '.';

const data = {
  name: 'username',
  label: 'Username',
  value: 'sergioalvarez',
  validationMessage: 'Username already exists',
};

describe('TextInput', () => {
  it('should match snapshot', async () => {
    // Arrange

    // Act
    const { container } = render(<TextInput {...data} />);

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('should set value', () => {
    // Arrange
    const expected = 'Firstname';

    // Act
    const { getByDisplayValue } = render(<TextInput value={expected} />);
    const container = getByDisplayValue(expected);

    // Assert
    expect(container).toHaveAttribute('value', expected);
  });

  it('should set name', () => {
    // Arrange
    const expected = 'firstname';

    // Act
    const { getByDisplayValue } = render(<TextInput value={'Firstname'} name={expected} />);
    const container = getByDisplayValue('Firstname');

    // Assert
    expect(container).toHaveAttribute('name', expected);
  });

  it('should display validation error', () => {
    // Arrange
    const expected = data.validationMessage;

    // Act
    const { getByText } = render(<TextInput value={data.value} validationMessage={expected} />);
    const container = getByText(expected);

    // Assert
    expect(container).toMatchInlineSnapshot(`
      <div
        class="inputField__error"
      >
        Username already exists
      </div>
    `);
  });
});
