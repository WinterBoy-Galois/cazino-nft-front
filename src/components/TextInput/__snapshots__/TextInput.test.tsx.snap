// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should match snapshot 1`] = `
<div>
  <div
    class="inputField__container"
  >
    <div
      class="inputField__wrapper inputField__wrapper--error"
    >
      <label
        class="inputField__label"
      >
        Username
      </label>
      <input
        autocomplete="off"
        class="inputField"
        name="username"
        type="text"
        value="sergioalvarez"
      />
    </div>
    <div
      class="inputField__error"
    >
      Username already exists
    </div>
  </div>
</div>
`;
